src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -113,7 +120,11 @@ subroutine get_fv3_regional_ensperts_run(this,en_perts,nelen,ps_bar)
   ! 
   ! READ ENEMBLE MEMBERS DATA
             if (mype == 0) write(6,'(a,a)') 'CALL READ_FV3_REGIONAL_ENSPERTS FOR ENS DATA with the filename str : ',trim(ensfilenam_str)
-            call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz) 
+            if ( l_use_dbz_directDA ) then
+               call this%general_read_fv3_regional_dirZDA(fv3_filename,ps,u,v,tv,rh,oz,ql,qi,qr,qs,qg,qnr,w)
```

Comment:
I suggest to think making the subroutine general_read_fv3_regional_dirZDA only reads in "ql,qi,qr,qs,qg,qnr,w".
the "general_read_fv3_regional" still works for all situation but "general_read_fv3_regional_dirZDA " can read extra fields if l_use_dbz_directDA is turned on. This is to avoid duplicated code in both subroutines.

Reply:
I modified subroutine ‘general_read_fv3_regional_dirZDA’ to read hydrometers and ‘w’ only according to your comments.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-86aea35b64364db7913dd1eb935d98f572e0601c925defcb5595748644029235R141-R144
https://github.com/NOAA-EMC/GSI/pull/142/files#diff-86aea35b64364db7913dd1eb935d98f572e0601c925defcb5595748644029235R578-R724